aptos-move/aptos-aggregator/src/delta_change_set.rs

```diff
@@ -76,27 +76,66 @@ impl DeltaOp {
         }
     }
 
-    /// Aggregates another delta into `self`.
-    pub fn merge_with(&mut self, other: DeltaOp) -> PartialVMResult<()> {
+    /// Applies self on top of previous delta, merging them together. Note
+    /// that the strict ordering here is crucial for catching overflows
+    /// correctly.
+    pub fn merge_onto(&mut self, previous_delta: DeltaOp) -> PartialVMResult<()> {
         use DeltaUpdate::*;
 
-        Ok(match (self.update, other.update) {
-            (Plus(value), Plus(other_value)) => {
-                let new_value = addition(value, other_value, self.limit)?;
-                self.update = Plus(new_value);
-            }
-            (Plus(value), Minus(other_value)) | (Minus(other_value), Plus(value)) => {
-                if value >= other_value {
-                    let new_value = subtraction(value, other_value)?;
-                    self.update = Plus(new_value);
-                } else {
-                    let new_value = subtraction(other_value, value)?;
-                    self.update = Minus(new_value);
+        Ok(match previous_delta.update {
+            Plus(value) => {
```

Comment:
we could call this prev_value, and the "other_value" would then be self_value or just value.

Reply:
agreed!